### PR TITLE
Group similar routes

### PR DESCRIPTION
**Pull request by steeevin88**
Made changes to group similar stops (ex. San Jose Diridon Caltrain Station Northbound, San Jose Diridon Caltrain Station Southbound)

Main change is that the Prediction dataclass holds a dictionary of destinations with corresponding times --> {destination_name, times[]}

stops part of data.yaml looks like this now:
```yaml
stops:
   - operator: SC
      name: 10th & Taylor
      id: 64995
grouped_stops:
    - operator: CT
      group_name: San Jose Diridon Caltrain Station
      ids:
        - 70261
        - 70262
 ```
 
 Sample output at port 8001 for just the grouped stops:
 ```json
{
  "ids": [70261, 70262],
  "name": "San Jose Diridon Caltrain Station",
  "predictions": [
    {
      "route": "L2",
      "destinations": {
        "San Francisco": [
          "2024-01-28T00:12:00Z",
          "2024-01-28T01:12:00Z"
        ],
        "Tamien": [
          "2024-01-28T00:40:00Z",
          "2024-01-28T02:40:00Z"
        ]
      }
    }
  ]
}

```

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -71,38 +70,51 @@ class Cache:
...
-    }
     new_stops = []
 
     # send post request for each agency's stop(s)
+    for stop in stops:
+        # check for stop grouping
+        if stop.get("group_name"):
+            group_name = stop.get("group_name")
+            stop_info = get_stop_predictions(stop.get("stops"), group_name)
+            new_stops.append(stop_info)
```
Comment (evanugarte) [inline app.py]:
```suggestion
        if stop.get("group_name"):
            group_name = stop.get("group_name")
            stop_info = get_stop_predictions(stop.get("stops", []), group_name)
            new_stops.append(stop_info)
            continue
```

File: app.py [hunk 2]
```diff
@@ -71,38 +70,51 @@ class Cache:
...
+            
+        # otherwise, get predictions for individual stop
+        else:
+            stop_name = add_suffix_to_name(stop)
+            stop_info = get_stop_predictions([stop], stop_name)
+            new_stops.append(stop_info)
```
Comment (evanugarte) [inline app.py]:
```suggestion
        stop_name = add_suffix_to_name(stop)
        stop_info = get_stop_predictions([stop], stop_name)
        new_stops.append(stop_info)
```

File: app.py [hunk 3]
```diff
@@ -71,38 +70,51 @@ class Cache:
...
             response = requests.get(PREDICTIONS_URL, params=params)
         logging.debug(f'511`s API response code was {response.status_code}')
+
         # unsuccessful request to 511 API
         if response.status_code != 200:
             MetricsHandler.api_response_codes.labels(response.status_code).inc() 
             logging.error(f"not parsing response because response code was {response.status_code}")
             time.sleep(10)
-            continue
+            return
```
Comment (evanugarte) [inline app.py]: why did we change this?

File: app.py (hunk 1 above)
Comment (steeevin88) [reply app.py]: resolved

File: app.py (hunk 2 above)
Comment (steeevin88) [reply app.py]: resolved

File: app.py (hunk 3 above)
Comment (steeevin88) [reply app.py]: `continue` caused an error when I tried doing it a different way but I forgot to change it back

resolved

**Review by evanugarte — APPROVED** [review]

**Pull request merged**